### ISSUE TITLE
ReadTheDocs -> GitHub Pages migration

### ISSUE THREAD

**Issue by renefritze** [pymor/pymor#863]
Just some stuff we need to remember/discuss: 
 1. Before we can fully migrate we need to include all (?) past doc builds for (last in major series) releases  in the pages space.
 1. ~~https for Gitlab Pages not tested/enabled yet~~
 1. Gitlab Pages has no nice JS "switch to version" menu. 
 1. ~~#862 needs to be merged~~

What else, @pymor/pymor-devs ?

**Comment by renefritze** [comment]
5. we should guard pages deploy with Gitlab CI's resource semaphores once they become available.

**Comment by renefritze** [comment]
> 1. we should guard pages deploy with Gitlab CI's resource semaphores once they become available.

This is actually now available in Gitlab 12.7 despite the Gitlab Issue not mentioning it explicitly. I will PR this in a bit.

**Comment by renefritze** [comment]
> > 1. we should guard pages deploy with Gitlab CI's resource semaphores once they become available.
> 
> This is actually now available in Gitlab 12.7 despite the Gitlab Issue not mentioning it explicitly. I will PR this in a bit.

Implemented in #882

**Comment by renefritze** [comment]
https is also now enabled, but due to some technical limitations between gitlab, WWU IT and letsencrypt still needs my manual intervention for certificate renewal every 90 days

**Comment by renefritze** [comment]
Using gitlab pages with the manual carrying over of past builds proved too unreliable. I've switched to serving https://github.com/pymor/docs as https://gh-docs.pymor.org
This docs git is updated via gitlab-ci

Remaining TODOS:
 1. put previous releases' docs into git
 1. css load problems on  https://gh-docs.pymor.org
 1. index is currently pointing to a list, should be "current" release, but see 1.

**Comment by renefritze** [comment]
https://github.com/pymor/pymor/pull/965 implements the pyMOR proper side of this. Everything else is an issue outside this repo:
https://github.com/pymor/docs/issues/2
https://github.com/pymor/docs/issues/3

**Comment by renefritze** [comment]
_Actually_ we still have some RTD specific stuff in our sphinx setup

**Comment by renefritze** [comment]
The remaining RTD stuff in config.py concerns only mocking, which is no longer needed.
I've just put RTD's versions of 0.4.2 and 0.5.2 into the docs repo and the docs index points to the 2019.2.0 release. 
The last thing to do to close this issue is to point docs.pymor.org away from RTD to the github pages setup.
Unless there's concerns (@pymor/pymor-devs) I shall proceed to do that 2020-07-02 sometime after 15:00 UTC.

**Comment by renefritze** [comment]
That will only come with the new sphinx theme.

**Comment by sdrave** [comment]
Ok, then we should probably use the new theme before migrating?

**Comment by renefritze** [comment]
I think the migration needs to have happened before the release. The version switcher is not important enough to block that. The new theme can be backported if it's not ready by release.

**Comment by sdrave** [comment]
So you would let the normal user only see the docs of the current stable versions when she visits docs.pymor.org?

**Comment by renefritze** [comment]
Yes.

**Comment by renefritze** [comment]
So I'm assuming 2-day silence as being OK with the plan and will do the change later today now for real.

**Comment by renefritze** [comment]
Switch is now complete, with https://rtd.pymor.org still serving the old site and https://docs.pymor.org serving the new.
